src/test/java/com/google/sps/CommentServiceTest.java

```diff
@@ -0,0 +1,90 @@
+package com.google.sps;
+
+import com.google.sps.servlets.CommentService;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+
+public final class CommentServiceTest {
+
+  CommentService commentService = new CommentService();
+
+  /**
+   * A list of Strings should be returned for any public video that has comments. 
+   */
+  @Test
+  public void publicVideoReturnsCommentsList() throws GeneralSecurityException, IOException {
+    String videoId = "v3whnmU619I"; // Public video that has some comments.
+    List<String> actual = commentService.getCommentsFromId(videoId);
+
+    Assert.assertFalse(actual.isEmpty());
+  }
+
+  /**
+   * Unlisted videos should be accessible to the API if the video id is inputted. List of comments
+   * should be returned for unlisted videos just like they would for a public video. 
+   */
+  @Test
+  public void unlistedVideoReturnsCommentsList() throws GeneralSecurityException, IOException {
+    String videoId = "9BUFUUhfXSg"; // Unlisted test video that has 2 comments. 
+    List<String> expected = Arrays.asList("test comment 2", "test comment 1");
+    List<String> actual = commentService.getCommentsFromId(videoId);
+
+    Assert.assertEquals(expected, actual);
```

Comment:
nit: I am not sure how "sort by relevance" is defined, but if "test comment 1" on your test video gets a lot of thumbs up, is the expected list in a different order and break your test?

Reply:
This is a really good point, and I think my logic for this test was flawed now that I think more about it. Originally I thought, since I unlisted this video, no one other than me could access it and be able to affect the comment section. But I failed to acknowledge that the video ID is publicly available on github so anyone could find it and break this test :/

A current solution may be using containsAll() and adjusting the video settings so only approved comments are posted. I feel like this isn't a very sufficient test though.

Reply:
It makes sense to me that by unlisting the video, people won't find the video as easily. And also I think this is less a coding/logic issue but more of an infrastructure one. I can't think of a good workaround either 🤷‍♀️

Reply:
This is a good example of why unit test should be self contained and not actually call real services. You should be mocking the response to something hard coded so the test doesn't become flaky.